### PR TITLE
Task undefined bug 86

### PR DESCRIPTION
**Pull request by gareththegeek**
Fix for #86 
Early out of step function if end condition met

### CODE REVIEW

File: simulator/tests/SimulatorTests.ts [hunk 1]
```diff
@@ -302,4 +302,13 @@ describe("Simulator", () => {
         expect(actual.warriorIndex).not.to.be.equal(mutated.warriorIndex);
         expect(actual.warriors.length).to.be.equal(0);
     });
+
+    it("Should not execute step if end condition met", () => {
```
Comment (dougajmcdonald) [inline simulator/tests/SimulatorTests.ts]: Yey, tests!

**Comment by dougajmcdonald** [comment]
NIce!

File: package.json [hunk 2]
```diff
@@ -65,6 +65,7 @@
     "clone": "^2.1.1",
     "core-js": "^2.5.1",
     "pubsub-js": "^1.5.7",
-    "underscore": "^1.8.3"
+    "underscore": "^1.8.3",
+    "user": "0.0.0"
```
Comment (dougajmcdonald) [inline package.json]: What's this?

Comment (gareththegeek) [reply package.json]: I have no idea!?

Comment (gareththegeek) [reply package.json]: I think my npm is haunted tonight

**Pull request merged**